File: src/bridge/appBootstrap.ts
import { invoke } from "@tauri-apps/api/core";

export type AppBootstrap = {
  name: string;
  version: string;
  platform: string;
  architecture: string;
};

export function getAppBootstrap(): Promise<AppBootstrap> {
  return invoke<AppBootstrap>("get_app_bootstrap");
}

File: src-tauri/src/domain/app_bootstrap.rs
use serde::Serialize;

#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct AppBootstrap {
    pub(crate) name: &'static str,
    pub(crate) version: &'static str,
    pub(crate) platform: &'static str,
    pub(crate) architecture: &'static str,
}
